dependencies.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import sys
+import subprocess
+import json
+
+# Set default logging level to INFO
+logging.basicConfig(level="INFO")
+
+# Check if current python version is 3.8
+if sys.version_info < (3, 8):
+    logging.fatal('McPy needs Python version 3.8.0 or higher to run! Current version is %s.%s.%s' % (
+        sys.version_info.major, sys.version_info.minor, sys.version_info.micro))
+    sys.exit(-4)
+
+# Make sure pip is installed
+logging.info("Making sure pip is installed...")
+subprocess.check_call([sys.executable, '-m', 'ensurepip'])
+logging.info("Pip is installed")
+
+# Make sure that dependencies are installed
+logging.info("Installing dependencies...")
+subprocess.check_call([sys.executable, '-m', 'pip', 'install', '-r', 'requirements.txt'])
+try:
+    subprocess.check_call(['git', 'init'])
+    subprocess.check_call(['git', 'submodule', 'init'])
+    subprocess.check_call(['git', 'submodule', 'update'])
+except:
+    logging.fatal("Git is not installed! Please install it!")
+logging.info("Dependencies installed")
```

Comment:
I would put the last line (line 29) before the except line (so insert between line 26 and 27) provided that the git submodules are required, if they are not, then further information should be given and the fatal converted into a warning.
I would also only catch the correct exception, although I do not know if there is a specific exception that is raised if an executable wasn't found on the path